src/server.rs

```diff
@@ -297,10 +302,81 @@ async fn upload(
     Ok(HttpResponse::Ok().body(urls.join("")))
 }
 
+#[derive(Serialize, Deserialize)]
+struct ListItem {
+    file_name: String,
+    file_size: u64,
+    expires_at: Option<String>,
+}
+
+// Returns a list of files
+#[get("/list")]
+async fn list(
+    request: HttpRequest,
+    config: web::Data<RwLock<Config>>,
+) -> Result<HttpResponse, Error> {
+    let config = config
+        .read()
+        .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?
+        .clone();
+
+    if !config.server.json_list_enabled.unwrap_or(false) {
+        Err(error::ErrorForbidden("json list is not enabled"))?;
```

Comment:
maybe this should more consistent with the `/version` endpoint.

e.g. adding a warning 

```suggestion
        log::warn!("server is not configured to expose list endpoint");
Err(error::ErrorForbidden("endpoint is not exposed"))?;
```

But it's just an idea.

Reply:
Agree with @tessus here.